src/dolphin/_show_versions.py

```diff
@@ -0,0 +1,104 @@
+"""Utility methods to print system info for debugging.
+
+Adapted from `rasterio.show_versions`,
+which was adapted from `sklearn.utils._show_versions`
+which was adapted from `pandas.show_versions`
+"""
+import importlib
+import platform
+import sys
+
+__all__ = ["show_versions"]
+
+
+def _get_sys_info():
+    """System information.
+
+    Returns
+    -------
+    dict:
+        system and Python version information
+    """
+    blob = [
+        ("python", sys.version.replace("\n", " ")),
+        ("executable", sys.executable),
+        ("machine", platform.platform()),
+    ]
+
+    return dict(blob)
```

Comment:
Not sure why we're creating a `list` initially and then converting it to a `dict`

```suggestion
    return {
        "python": sys.version.replace("\n", " "),
        "executable": sys.executable,
        "machine": platform.platform(),
    }
```

Reply:
changing these quirks (coming from rasterio.show_versions`)